### ISSUE TITLE
Icons accessibility

### ISSUE THREAD

**Issue by vlajna95** [squidfunk/mkdocs-material#2885]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [x] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... the documentation does not mention anything about my idea
- [X] ... there are no open or closed issues that are related to my idea

### Description

The icons are not visible to the screen reader programs. Is there a way to add a alt/title attributes to the element which renders the icon? It should contain, for example, the name of the icon (e.g. `:material-check:`).

### Use Cases

It's OK for the icons to be invisible when they are used merely as a decoration, but we often use them to point something out, and I think the effort to fix this accessibility issue is relatively small compared to the benefits for blind and visually impaired users.

**Comment by squidfunk** [comment]
Thanks for suggesting! Icons are rendered by the [Emoji extension](https://facelessuser.github.io/pymdown-extensions/extensions/emoji/), which is an upstream dependency of Material for MkDocs. Maybe you could discuss possible improvements with @facelessuser, who is the maintainer of this project.

**Comment by facelessuser** [comment]
I wonder if this is specific to emoji in general or just the icon support we add for Material.

**Comment by facelessuser** [comment]
Pretty sure this is specific to Material icons only which would point to this repository: https://github.com/facelessuser/mkdocs-material-extensions. Yes it piggy backs on the emoji extension in pymdownx, but it is generating the icons a bit differently than it does the emoji images as it is inlining SVG from the Material package. So, maybe create an issue there and we can add a title on the span wrapping the SVG images or something.

**Comment by squidfunk** [comment]
Thanks for your take on this. I thought it might be solvable more general, because emojis are rendered kind of similarily and also suffer from the same accessibility issues, don't they?

Regarding the icons: unfortunately there's no title attached to the `@mdi/svg` icons we use, see for example [this icon](https://github.com/Templarian/MaterialDesign-SVG/blame/master/svg/account-arrow-left.svg). The only text in the SVG is the `id`, which equals the file name. This is just a random sample, but I think it's pretty safe to assume that this is the same for all other icons. Thus, we could infer the title from the file name, which might work for English but break down for other languages (because we only have English), possibly tripping screen readers up. I'm German, and try the Material for MkDocs site with VoiceOver from time to time to improve on accessibility, and mixing English and German already sounds pretty weird.

I'm undecided if setting the icon name as a title would be a real improvement. Maybe we could use `attr_list` to allow setting a title explicitly on an icon? Something similar to:

```
Click the :material-account-arrow-left:{ title="go back" } icon.
```

**Comment by facelessuser** [comment]
Normal emoji have titles. I verified this. Special icons, like Material icons, do not. Now, whether titles are actually helpful...🤷🏻.

I'm also not aware if titles alone are sufficient for screen readers. I know normal emojis are generated with titles, but I've never run a screen reader on them. to verify whether it will read them.

Further, I know the current `attr_list` extension won't add a `title` to SVG as demonstrated above, it will only attach an attribute. Could we provide some workaround? Possibly, but it won't out of the box. Also, it won't add the attribute to the SVG, but probably to the wrapper span.

**Comment by squidfunk** [comment]
> Also, it won't add the attribute to the SVG, but probably to the wrapper span.

Hmm, maybe using `aria-label` on the wrapper span might be sufficient then.

**Comment by facelessuser** [comment]
Something to look into.
